src/content/en/updates/2018/03/present-web-pages-to-secondary-attached-displays.md

```diff
@@ -0,0 +1,268 @@
+instance and wait for the promise to resolve once user has selected a
+presentation display (.e.g. a secondary attached display in our use case).
+
+    function onPresentButtonClick() {
+      presentationRequest.start()
+      .then(connection => {
+        console.log('Connected to ' + connection.url + ', id: ' + connection.id);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+    }
+
+List presented to the user may also include remote endpoints such as Chromecast
+devices if you’re connected to a network advertising some.
+
+<figure>
+  <img src="/web/updates/images/2018/03/present-web-pages-to-secondary-attached-displays/picker.png"
+    alt="Presentation Display Picker"/>
+</figure>
+
+When promise resolves, the web page at the `PresentationRequest` object URL is
+presented to the attached display user has previously picked. And voila!
+
+We can now go further and monitor "close" and "terminate" events quite easily
+as you can see below. Note that it is possible to reconnect to a "closed"
+`presentationConnection` with `presentationRequest.reconnect(presentationId)`
+where `presentationId` is the ID of the previous `presentationRequest` object.
+
+    function onCloseButtonClick() {
+      // Disconnect presentation connection but will allow reconnection.
+      presentationConnection.close();
+    }
+    
+    presentationConnection.addEventListener('close', function() {
+      console.log('Connection closed.');
+    });
+    
+    
+    function onTerminateButtonClick() {
+      // Stop presentation connection for good.
+      presentationConnection.terminate();
+    }
+    
+    presentationConnection.addEventListener('terminate', function() {
+      console.log('Connection terminated.');
+    });
+
+## Communicate with the page
+
+That’s nice but how do I pass messages between my controller page (the one
+we’ve just created) and the receiver page (the one we’ve passed to the
+`PresentationRequest` object)?
+
+First, let’s retrieve existing connections on the receiver page with
+`navigator.presentation.receiver.connectionList` and listen to incoming
+connections as shown below. 
+
+    // Receiver page
+    
+    navigator.presentation.receiver.connectionList
+    .then(list => {
+      list.connections.map(connection => addConnection(connection));
+      list.addEventListener('connectionavailable', function(event) {
+        addConnection(event.connection);
+      });
+    });
+    
+    function addConnection(connection) {
+    
+      connection.addEventListener('message', function(event) {
+        console.log('Message: ' + event.data);
+        connection.send('Hey controller! I just received a message.');
+      });
+    
+      connection.addEventListener('close', function(event) {
+        console.log('Connection closed!', event.reason);
+      });
+    }
+
+A connection receiving a message will fire a "message" event you can listen to.
```

Comment:
'will fire' > 'fires'
'listen to' > 'listen for'

Reply:
Done.